catalog-rest-service/src/main/java/org/openmetadata/catalog/security/policyevaluator/PolicyEvaluator.java

```diff
@@ -29,21 +32,44 @@
  *
  * <p>- {@link org.openmetadata.catalog.Entity} (object) on which to operate on.
  */
+@Slf4j
 public class PolicyEvaluator {
 
-  private final Rules rules;
-  private final RulesEngine rulesEngine;
+  private PolicyRepository policyRepository;
+  private AtomicReference<Rules> rules;
+  private RulesEngine rulesEngine;
 
-  public PolicyEvaluator(List<org.openmetadata.catalog.entity.policies.accessControl.Rule> rules) {
-    this.rules = new Rules();
-    rules.stream()
+  // Eager initialization of Singleton since PolicyEvaluator is lightweight.
+  private static final PolicyEvaluator policyEvaluator = new PolicyEvaluator();
```

Comment:
@harshach  Singleton init here

Reply:
This may run slow down server start when we have lot of policies. But shouldn't be a big deal

Reply:
With current implementation, the actual loading happens in a different method call from the constructor

The constructor intelligently loads empty rules `rules = new AtomicReference<>(new Rules());`


Besides this there is one last bit of optimization I'd like to do on this (which I'll come back to after I get couple more PRs out of the way to unblock the UI work)